cmd/minishift/cmd/util/util.go

```diff
@@ -112,3 +115,44 @@ func IsValidProfileName(profileName string) bool {
 	rr := regexp.MustCompile(`^[a-zA-Z0-9]+[\w+-]*$`)
 	return rr.MatchString(profileName)
 }
+
+// RunAndGetExitCode returns the exit code of a command
+func RunAndGetExitCode(command string, args ...string) int {
+	const defaultFailedCode = 1
+	cmd := exec.Command(command, args...)
+	err := cmd.Run()
+	if err != nil {
+		// the command failed to execute
+		// try to get the system exit code
+		if exitError, ok := err.(*exec.ExitError); ok {
+			ws := exitError.Sys().(syscall.WaitStatus)
+			// ExitStatus has the exit code
+			return ws.ExitStatus()
+		}
+		return defaultFailedCode
+	}
+	// success, exit code should be 0 if go is ok
+	ws := cmd.ProcessState.Sys().(syscall.WaitStatus)
+	return ws.ExitStatus()
+}
+
+// CommandExecutesSuccessfully returns true if the command executed based on the exit code
+func CommandExecutesSuccessfully(cmd string, args ...string) bool {
+	// check in the path if cmd	is present
+	path, err := exec.LookPath(cmd)
+	if err != nil {
+		return false
+	}
+	fi, _ := os.Stat(path)
+	if fi.Mode()&os.ModeSymlink != 0 {
```

Comment:
What is the purpose of this? Why even have this check?

Reply:
To check if the binary in the `PATH` is a symlink, and check if the target exists.